### PR TITLE
Fix offline stream settings

### PR DESCRIPTION
**Pull request by CasperMok**
* Set settings to stream and global in the stream view mode
* Remove stream detail from subscribed state

### CODE REVIEW

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift [hunk 1]
```diff
@@ -93,7 +93,9 @@ public struct StreamingScreen: View {
                 // TODO: Add title
             }
             ToolbarItem(placement: .navigationBarTrailing) {
-                SettingsButton(isShowingSettingsScreen: $isShowingSettingsScreen)
+                if case .stream = viewModel.settingsManager.mode {
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift:96]: Would we have a scenario where the streaming screen is presented in `.global` mode ?

Comment (CasperMok) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift:96]: The stream is offline at the beginning.

Comment (aravind-raveendran) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift:96]: Got it. Would it be meaningful to show settings even when stream is not live ? That would probably mean we set the `setActiveSetting(streamId` as soon as we have `connect` successful but still waiting for subscribe to return success ?

I remember - Android TV app shows settings even when the subscription is not live.

Comment (CasperMok) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift:96]: This is what I talked about with you before. We can't get the `streamID` until the view model gets the .subscribed(). Or We may assume the `streamID`, make some changes to the connection coordinator, and how we pass the `streamID` to the settings manager

**Review by aravind-raveendran — APPROVED** [review]

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift (hunk 1 above)
Comment (CasperMok) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift]: Public private(set) activeStreamDetail in coordinator

Comment (aravind-raveendran) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift]: Is it safe to now remove the following lines of code in function -> `private func updateState(from sources: [StreamSource], streamDetail: StreamDetail, settings: StreamSettings)` ?

// When retreiving sources for the first time
        if self.sources.isEmpty {
            // Update settings manager with the current stream information
            settingsManager.setActiveSetting(for: .stream(streamID: streamDetail.streamId))
        }

Comment (CasperMok) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift]: Right, forgot. Thanks

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift [hunk 2]
```diff
@@ -105,6 +105,7 @@ extension StreamingScreen {
     func endStream() {
         Task {
             await viewModel.endStream()
+            viewModel.settingsManager.setActiveSetting(for: .global)
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift:108]: we can do it in the `endStream` function in `StreamViewModel` ?

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift (hunk 1 above)
Comment (CasperMok) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift]: Do you want me to remove the streamDetail from .subscribed() as well?

Comment (aravind-raveendran) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift]: That would be great. Currently I exposed that value in `.subscribed` state with the intention of using it for `Settings`. 

But I can pick it up if it increases the scope of this PR ?

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift (hunk 2 above)
Comment (CasperMok) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Components/StreamingScreen.swift:108]: Rite

**Pull request merged**